api/client/src/main/java/org/projectnessie/client/http/HttpRequest.java

```diff
@@ -31,7 +35,16 @@ public abstract class HttpRequest
   protected String accept = "application/json; charset=utf-8";
 
   protected HttpRequest(HttpRuntimeConfig config) {
-    this.uriBuilder = new UriBuilder(config.getBaseUri());
+    this(config, config.getBaseUri());
+  }
+
+  protected HttpRequest(HttpRuntimeConfig config, URI baseUri) {
+    requireNonNull(baseUri, "Base URI cannot be null");
+    checkArgument(
+        "http".equals(baseUri.getScheme()) || "https".equals(baseUri.getScheme()),
```

Comment:
Maybe `equalsIgnoreCase`? Per [rfc3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) implementations should accept either case.